bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/NavigationImpl.java

```diff
@@ -254,8 +255,9 @@ public String getExportedType() {
      */
     private NavigationItem createNavigationItem(@NotNull final Page page, @NotNull final List<NavigationItem> children) {
         int level = page.getDepth() - (this.getNavigationRoot().getDepth() + structureStart);
+        boolean current = checkCurrent(page);
         boolean selected = checkSelected(page);
```

Comment:
This causes `checkCurrent` to be called twice in a row, I'd rather we pass the value of `current` to `checkSelected`.